app/scripts/lib/ppom/ppom-middleware.ts

```diff
@@ -14,7 +14,7 @@ import { SIGNING_METHODS } from '../../../../shared/constants/transaction';
 import PreferencesController from '../../controllers/preferences-controller';
 import { AppStateController } from '../../controllers/app-state';
 import { LOADING_SECURITY_ALERT_RESPONSE } from '../../../../shared/constants/security-provider';
-import { getProviderConfig } from '../../../../ui/ducks/metamask/metamask';
+import { getCurrentChainId } from '../../../../ui/selectors';
```

Comment:
For a set of different reasons, some time ago there was an effort to isolate and keep the ui code decoupled from the background code. Anything that was to be used by both ui and background should be moved into the `shared` directory.

I would suggest moving any ui imports in the background (`app` directory) to the `shared` dir.

That said, it also feels prone to future errors re-using a ui selector, that (in theory) expects to have the entire redux state available to it.

Per redux docs:
```
Selectors, on the other hand, usually are written to take the entire Redux root state as their argument. This means that they have to know where in the root state this slice's data is kept, such as state.todos, even though that's not really defined until the root reducer is created (typically in the app-wide store setup logic).
```

Although passing the networkController state as metamask state currently works, it adds unexpected complexity to any future change/update to the selector (as the developer expectations are not aligned to the selector usecases).
I would suggest that is preferable to not rely on the ui selectors even if it means duplicating the selector logic.
What do you think?

Reply:
My first point would be that the changes in this file are not actually necessary and could be kept as they were.  But your point would remain as it was previously explicitly passing state to a selector.

That happened as part of the [v20 network controller upgrade](https://github.com/MetaMask/metamask-extension/pull/26150), where `providerConfig` was removed from state.  From that point, there was no longer a way to directly grab the current chain id from state.  But many places still need to do so.

The advice I received for doing that upgrade was to shim `getProviderConfig` to calculate and return equivalent data.  And to use that shim *even* for places that were accessing state directly like this, and weren't UI selectors.  There's a bunch more background places like this including MetamaskController and others.

That still applies in this v21 upgrade, with the logic changed under the hood.  It's still nice to encapsulate this logic as the network state evolves.  But I guess you're just suggesting it gets duplicated into a `shared` directory. 

I don't oppose it, but also don't know the historic reasons to advocate for the separation.  Since the shims are in place from a previous upgrade, I'd suggest we consider tackling it in a future refactor.

Reply:
Yes, my opinion is that we should duplicate the UI selector and for the background to use.. The minimum that we should do is having the selector moved into the shared directory.

There are multiple reasons to have the UI decoupled from the Background/service worker and vice-versa (for example both have different bundles, and our build system currently does not support tree shaking. If we start coupling UI and background, besides a more complex/entabgled codebase, we are also producing unnecessarly bigger bundles for each). But, specifically for UI selectors, the main reason is the state that is expected. On the UI we should expect the entire redux state to be available (which is not the case on the background). And just because currently it converges, it does not mean that in the Future both UI and Background state cannot evolve in ways that make this selector incompatible. That is why, moving it to the shared directory would explicitly state to other devs that this selector is used on both UI and background.

That said, I agree that we can do those changes in a follow up PR as you suggested.